### PR TITLE
Update the prepare_prebuilt_workers to be able to specify repo.

### PR DESCRIPTION
**Pull request by wanlin31**
The Pr provides a way for user to specify which repo they want to
build the test worker image from.

### CODE REVIEW

File: tools/cmd/prepare_prebuilt_workers/main.go [hunk 1]
```diff
@@ -105,41 +112,51 @@ func main() {
 	for _, pair := range languagesSelected {
 		split := strings.Split(pair, ":")
 
-		if len(split) != 2 || split[len(split)-1] == "" {
-			log.Fatalf("Input error in language and gitref selection, please follow the format as language:gitref, for example: c++:<commit-sha>")
+		if len(split) < 2 || split[len(split)-1] == "" {
+			log.Fatalf("Input error in language and gitref selection, please follow the format as language:repository:gitref, for example: c++:grpc/grpc/<commit-sha>")
```
Comment (paulosjca) [inline tools/cmd/prepare_prebuilt_workers/main.go:116]: "selection. Please follow the format language:gitref or language:repository:gitref, for example c++:master or c++:grpc/grpc:master".

File: tools/cmd/prepare_prebuilt_workers/main.go [hunk 2]
```diff
@@ -105,41 +112,51 @@ func main() {
...
 
 			// build image
-			log.Println(fmt.Sprintf("building %s image", lang))
-			// TODO(jtattermusch): allow setting REPOSITORY arg to allow building from user forks e.g. https://github.com/USER_NAME/grpc.git
+			log.Printf("building %s image\n", lang)
 			buildCommandTimeoutSeconds := 30 * 60 // 30 mins should be enough for all languages
-			buildDockerImage := exec.Command("timeout", fmt.Sprintf("%ds", buildCommandTimeoutSeconds), "docker", "build", dockerfileLocation, "-t", image, "--build-arg", fmt.Sprintf("GITREF=%s", gitRef), "--build-arg", fmt.Sprintf("BREAK_CACHE=%s", uniqueCacheBreaker))
+			buildDockerImage := exec.Command("timeout", fmt.Sprintf("%ds", buildCommandTimeoutSeconds), "docker", "build", dockerfileLocation, "-t", image, "--build-arg", fmt.Sprintf("GITREF=%s", spec.Gitref), "--build-arg", fmt.Sprintf("BREAK_CACHE=%s", uniqueCacheBreaker))
+			if spec.Repo != "" {
+				//cmd = fmt.Sprintf("docker build %s -t %s --build-arg REPOSITORY=%s --build-arg GITREF=%s --build-arg BREAK_CACHE=VAR", dockerfileLocation, image, spec.Repo, spec.Gitref)
```
Comment (paulosjca) [inline tools/cmd/prepare_prebuilt_workers/main.go:156]: You should remove this extra comment.

File: tools/README.md [hunk 3]
```diff
@@ -98,13 +98,27 @@ bin/prepare_prebuilt_workers \
      -r containers/pre_built_workers
 ```
 
-This builds `cxx` and `go` images contain workers built from the commit/branch
-we wish to test.
+This builds `cxx` and `go` images contain workers built from the specific
+commit/branch we wish to test.
+
+You may also specify the repo where the source code is pulled from by specifying
+the repo, the following command would pull the source code from a forked repo
```
Comment (paulosjca) [inline tools/README.md:105]: "repo. The"

File: tools/README.md [hunk 4]
```diff
@@ -98,13 +98,27 @@ bin/prepare_prebuilt_workers \
...
+to build `cxx` and `go` images.
+
+```shell
+bin/prepare_prebuilt_workers \
+     -l cxx:<user-name>/grpc:master \
```
Comment (paulosjca) [inline tools/README.md:110]: Here I would use "cxx:${USER}/grpc:master" and "go:${USER}/grpc-go:master" to make the examples more concrete.

File: tools/README.md [hunk 5]
```diff
@@ -98,13 +98,27 @@ bin/prepare_prebuilt_workers \
...
+This builds `cxx` and `go` images contain workers built from the specific
+commit/branch we wish to test.
+
+You may also specify the repo where the source code is pulled from by specifying
+the repo. The following command pulls the source code from forked repos
+`${USER}/grpc` and `${USER}/grpc-go` to build cxx and go images
+
+```shell
+bin/prepare_prebuilt_workers \
+     -l cxx:${USER}/grpc:master \
```
Comment (paulosjca) [inline tools/README.md:110]: Please "" around the strings in line 110 and 111.

File: tools/cmd/prepare_prebuilt_workers/main.go [hunk 6]
```diff
@@ -105,41 +112,52 @@ func main() {
 	for _, pair := range languagesSelected {
```
Comment (paulosjca) [inline tools/cmd/prepare_prebuilt_workers/main.go:112]: Please use `SplitN` and rename `pair` according to earlier comment.

File: tools/README.md [hunk 7]
```diff
@@ -98,13 +98,27 @@ bin/prepare_prebuilt_workers \
...
+This builds `cxx` and `go` images contain workers built from the specific
+commit/branch we wish to test.
+
+You may also specify the repo where the source code is pulled from by specifying
+the repo. The following command pulls the source code from forked repos
+`${USER}/grpc` and `${USER}/grpc-go` to build cxx and go images
+
+```shell
+bin/prepare_prebuilt_workers \
+     -l cxx:"${USER}"/grpc:master \
```
Comment (paulosjca) [inline tools/README.md:110]: Please put the "" around the whole string, "cxx:${USER}/grpc:master" and "go:${USER}/grpc-go:master".

Comment (wanlin31) [reply tools/README.md]: The reason is to prevent splitting?  -- "An argument enclosed in double quotes presents itself as a single word" Thank you.

**Review by paulosjca — APPROVED** [review]

**Pull request merged**